src/com/madalinb/coffeeshop/data/CoffeeShop.kt

```diff
@@ -0,0 +1,7 @@
+package com.madalinb.coffeeshop.data
+
+data class CoffeeShop(var name: String, var x: Double, var y: Double) {
+    fun distanceTo(x2: Double, y2: Double): Double {
```

Comment:
Why does the method accept x and y? What alternatives do we have and how does it compare to this one?

Reply:
In this case, because the documentation says that we have to assume that all coordinates lie on a plane, we are talking about Cartesian coordinates. In a real world where the Earth is not flat (as some people really belive), but round, the coordinates should be latitude and longitude, and the formula for calculating the distance would be different.

Reply:
I think @pirvudoru is refering here to a code smell called [primitive obsesion](https://refactoring.guru/smells/primitive-obsession) you are inferring that 2 doubles are coordinates, but we are asking if there is a better way to do this.

Reply:
Moved the function into a more appropriate class.